omod/src/main/java/org/openmrs/module/fhir/providers/RestfulEncounterResourceProvider.java

```diff
@@ -68,6 +71,18 @@ public Encounter getResourceById(@IdParam IdDt theId) {
 		return encounterResource.searchEncountersById(id);
 	}
 
+	/**
+	 * Search encounters by patient identifier
+	 *
+	 * @param identifier object containing the patient identifier
+	 */
+	@Search()
+	public List<Encounter> searchEncountersByPatientIdentifier(
+		@RequiredParam(name=Encounter.SP_PATIENT, chainWhitelist= {Patient.SP_IDENTIFIER}) ReferenceParam identifier
```

Comment:
Here is a sample GET request which I tested on localhost. 
[localhost:8080/openmrs/ws/fhir/Encounter?patient.identifier=101-6](http://localhost:8080/openmrs/ws/fhir/Encounter?patient.identifier=101-6)

I followed dynamic approach, because @surangak [suggested that it is more readable](https://issues.openmrs.org/browse/FM-15?focusedCommentId=217920&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-217920).

Should I use static chains ? Then according to my understanding, code should change as 
```java
public List<Encounter> searchEncountersByPatientIdentifier(
		@RequiredParam(name=Encounter.SP_SUBJECT+'.'+Patient.SP_IDENTIFIER) TokenParam identifier
	){
```
Is it (not tested) ?